fem/gridfunc.cpp

```diff
@@ -4550,4 +4550,409 @@ GridFunction *Extrude1DGridFunction(Mesh *mesh, Mesh *mesh2d,
    return sol2d;
 }
 
+KDTreeNodalProjection::KDTreeNodalProjection(GridFunction& dest_)
+{
+   dest=&dest_;
+   FiniteElementSpace* space=dest->FESpace();
+   Mesh* mesh=space->GetMesh();
+   int dim=mesh->SpaceDimension();
+
+   if (dim==2) { kdt2D=new KDTree2D(); kdt3D=nullptr;}
+   if (dim==3) { kdt3D=new KDTree3D(); kdt2D=nullptr;}
+   std::vector<bool> indt; indt.resize(space->GetVSize()/space->GetVDim());
+   for (size_t i=0; i<indt.size(); i++) {indt[i]=true;}
```

Comment:
Why not `std::vector<bool> indt(space->GetVSize()/space->GetVDim(), true);` instead of lines 4562-4563.

Reply:
corrected